scls-format/src/Cardano/SCLS/Internal/Serializer/Reference/Dump.hs

```diff
@@ -130,14 +135,7 @@ data ManifestInfo = ManifestInfo
   }
 
 instance Semigroup ManifestInfo where
-  (ManifestInfo a) <> (ManifestInfo b) = ManifestInfo (Map.unionWith combine a b)
-   where
-    combine x y =
-      NamespaceInfo
-        { namespaceEntries = namespaceEntries x + namespaceEntries y
-        , namespaceChunks = namespaceChunks x + namespaceChunks y
-        , namespaceHash = namespaceHash y -- TODO: combine hashes properly
-        }
+  (ManifestInfo a) <> (ManifestInfo b) = ManifestInfo (Map.union a b)
```

Comment:
We can write it like this because we assume that each namespace only appears once in a stream, right?

Reply:
Yes, `dump` takes normalised stream where we have each namespace only once in a serialised order.
Though I think we may want to make it more explicit by using either `First` or `Last` wrapper, let me think a bit especially if we can write it in a way that deltas will work as well